core/src/main/java/io/questdb/std/IntStack.java

```diff
@@ -72,13 +72,16 @@ public int pop() {
     public int pollLast() {
         final int[] es = elements;
         final int t;
-        int e = es[t = dec(tail, es.length)];
+        final int e = es[t = dec(tail, es.length)];
         tail = t;
+        es[t] = noEntryValue;
         return e;
     }
 
-    static int dec(int i, int modulus) {
-        if (--i < 0) i = modulus - 1;
+    private  int dec(int i, int modulus) {
+        if (head != tail && --i < 0) {
+            i = modulus - 1;
```

Comment:
nit: we could simply use `mask` instead of `modulus - 1` here.

Reply:
done: https://github.com/questdb/questdb/pull/2037/commits/f38e19731353ed9d2bf6ce4422741f04eabe7873